akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/BehaviorSetup.scala

```diff
@@ -70,8 +74,11 @@ private[akka] final class BehaviorSetup[C, E, S](
 
   val persistence: Persistence = Persistence(context.system.toClassic)
 
-  val journal: ClassicActorRef = persistence.journalFor(settings.journalPluginId)
-  val snapshotStore: ClassicActorRef = persistence.snapshotStoreFor(settings.snapshotPluginId)
+  val journal: ClassicActorRef =
+    persistence.journalFor(settings.journalPluginId, settings.journalPluginConfig.getOrElse(ConfigFactory.empty))
```

Comment:
Thanks for this PR, I agree that it would be a nice feature. Before merging I would like to see that we can support it all the way in the r2dbc plugin, which currently isn't using that config. I recall we had problems and gave up in the Cassandra plugin https://github.com/akka/akka-persistence-cassandra/pull/694

It's not only about the journal and snapshot. It should work with the queries too.

Reply:
Here is the PR for Pekko I was planning to port to Akka too - https://github.com/apache/pekko-projection/pull/225. The r2dbc plugin would just piggyback off that, IIUC. Is this fine?

Reply:
Aha, I think I finally understood what you had in mind with potential issues - my comment about projections was not relevant for that.

Reply:
My initial thinking is that journal plugin should simply not use the query plugin - the code doing the replay could be shared between journal and query plugins without the need for the former to depend on latter. @patriknw Does that make sense?